cssom-view-1/Overview.bs

```diff
@@ -1070,21 +1070,16 @@ result of running these steps:
         <dt><a>caret range</a>
         <dd>null
     </dl>
-1. Otherwise, return a <a>caret position</a> where the
-    <a>caret range</a> is a collapsed
-    {{Range}} object for the position
-    where the text insertion point indicator would have been inserted when applying the <a>transforms</a> that apply to the descendants of the
-    <a>viewport</a>, and the other properties are set as follows:
-
-    <dl>
-        <dt><a>caret node</a>
-        <dd>The [=range/start node=]
-            of the <a>caret range</a>.
-
-        <dt><a>caret offset</a>
-        <dd>The [=range/start offset=] of
-            the <a>caret range</a>.
-    </dl>
+1. Otherwise:
+
+    1. Let <var>caretRange</var> to be a collapsed {{Range}} object for the position where the text insertion point indicator would have been inserted when applying
+        the <a>transforms</a> that apply to the descendants of the <a>viewport</a>.
+    1. Let <var>startNode</var> be the [=range/start node=] of the <var>caretRange</var>, and let <var>startOffset</var> be the [=range/start offset=] of the <var>caretRange</var>.
+    1. While <var>startNode</var> is a [=node=], <var>startNode</var>'s [=tree/root=] is a [=shadow root=], and <var>startNode</var>'s [=tree/root=] is not a [=shadow-including inclusive ancestor=] of any of <var>shadowRoots</var>, repeat these steps:
+        1. Set <var>startOffset</var> to [=tree/index=] of <var>startNode</var>'s [=tree/root=]'s [=host=].
+        1. Set <var>startNode</var> to <var>startNode</var>'s [=tree/root=]'s [=host=]'s [=tree/parent=].</li>
+    1. Set <var>caretRange</var>'s [=range/start node=] and [=range/end node=] to <var>startNode</var>, and set <a>caret range</a>'s [=range/start offset=] and [=range/end offset=] to <var>startOffset</var>.
+    1. Return a <a>caret position</a> whose <a>caret node</a> is set to <var>startNode</var>, <a>caret offset</a> is set to <var>startOffset</var>, and <a>caret range</a> is set to <var>caretRange</var>.
```

Comment:
Could we create a "collapsed" `CaretPosition` upfront, instead of a range, and adjust the `offsetNode` and `offset` as needed?

Reply:
Updated to use general range concept instead of a live range in latest iteration.

Reply:
Isn't just using `CaretPosition` simpler? A range has two endpoints, which both need to be described like above, even though only one is really needed. Using a position concept would make this simpler.

Reply:
I think they are about the same in terms of expression simplicity. We still need to create a `Range` object for `caret range` in the last step.

I've updated the spec in latest iteration to let a position instead of a range to indicate text insertion point in first step.